job-engine/jbatch/src/main/java/org/eclipse/kapua/job/engine/jbatch/JobEngineServiceJbatch.java

```diff
@@ -189,4 +190,34 @@ public void stopJob(KapuaId scopeId, KapuaId jobId) throws KapuaException {
             throw new JobStopppingException(e, scopeId, jobId);
         }
     }
+
+    @Override
+    public void cleanJobData(KapuaId scopeId, KapuaId jobId) throws KapuaException {
+        //
+        // Argument Validation
+        ArgumentValidator.notNull(scopeId, "scopeId");
+        ArgumentValidator.notNull(jobId, "jobId");
+
+        //
+        // Check Access
+        AUTHORIZATION_SERVICE.checkPermission(PERMISSION_FACTORY.newPermission(JOB_DOMAIN, Actions.execute, scopeId));
+
+        //
+        // Check existence
+        Job job = JOB_SERVICE.find(scopeId, jobId);
+        if (job == null) {
+            throw new KapuaEntityNotFoundException(Job.TYPE, jobId);
+        }
+
+        //
+        // Check job not running
+        if (JbatchDriver.isRunningJob(scopeId, jobId)) {
+            throw new JobNotRunningException(scopeId, jobId);
```

Comment:
Opposite exception 😅  

Job is running so `JobRunningException` must be thrown